pkg/virt-launcher/virtwrap/network/podnic_test.go

```diff
@@ -0,0 +1,157 @@
+package network
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"runtime"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/golang/mock/gomock"
+	"github.com/vishvananda/netlink"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	kubevirtv1 "kubevirt.io/client-go/api/v1"
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/kubevirt/pkg/network/cache"
+	"kubevirt.io/kubevirt/pkg/network/cache/fake"
+	netdriver "kubevirt.io/kubevirt/pkg/network/driver"
+	neterrors "kubevirt.io/kubevirt/pkg/network/errors"
+	"kubevirt.io/kubevirt/pkg/network/infraconfigurators"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+var _ = Describe("podNIC", func() {
```

Comment:
I know this commit includes just moving the already existing tests to here. But those tests are not covering even 30% of the code in podNIC. Added some examples on the review comments, but there is much more.
Is there a plan to add more tests in a follow-up PR?

Reply:
Yes, we will increase the coverage when things get moved to `pkg/network` but moving to it's own podnic.go and refactoring the current tests is big enough for this PR.